libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/headerSection.spec.ts

```diff
@@ -0,0 +1,110 @@
+    describe("Default language", () => {
+        beforeEach(() => {
+            cy.login();
+
+            Navigation.visit("dashboard/header");
+            editMode.isInEditMode();
+            insightCatalog.waitForCatalogReload();
+        });
+
+        it("(SEPARATE) can update header for all sections", () => {
+            let i = 0;
+            let t = 0;
+            cy.fixture("headerDataTest").then((data) => {
+                data["DataTest"].forEach((result: { sectionName: string }) => {
+                    new LayoutRow(i)
+                        .getHeader()
+                        .setTitle(result.sectionName)
+                        .setDescription(result.sectionName);
+                    i += 1;
+                });
+            });
+
+            editMode.save(); //save all headers after input
+
+            cy.fixture("headerDataTest").then((data) => {
+                data["DataTest"].forEach((result: { sectionName: string }) => {
+                    if (t === 0) {
+                        new LayoutRow(t).getHeader().hasTitle(false).hasDescription(false);
+                    } else {
+                        new LayoutRow(t)
+                            .getHeader()
+                            .hasTitleWithText(result.sectionName)
+                            .hasDescriptionWithText(result.sectionName);
+                    }
+                    t += 1;
+                });
+            });
+        });
+
+        it("Header is removed after latest insight is deleted from a section", () => {
+            layoutRow_01.hasCountOfHeaderSection(3);
+            new Widget(0).removeVizWidget();
+
+            layoutRow_01.hasCountOfHeaderSection(2);
+        });
+    });
+
+    describe("Localization", () => {
+        before(() => {
+            cy.login();
+
+            Navigation.visit("dashboard/header-localization");
+            //cy.wait(10000); // need time to load localization config
+            editMode.isInEditMode();
+            insightCatalog.waitForCatalogReload();
+        });
+        it("Header placeholder should be translated", () => {
+            layoutRow_01
+                .getHeader()
+                .setTitle(" ")
+                .setDescription(" ")
+                .hasPlaceholderTitle("Ajouter un titre ici...")
+                .hasPlaceholderDescription("Ajouter une description ici...");
+        });
+
+        it("(SEPARATE) limitation of title, description", () => {
+            cy.fixture("headerDataTest").then((data) => {
+                const title = data["LimitTexts"].title;
+                const desc = data["LimitTexts"].description;
+                const headerRow_01 = layoutRow_01.getHeader();
+                const headerRow_02 = layoutRow_02.getHeader();
+
+                headerRow_01.setTitle(title);
+                headerRow_02.setDescription(desc);
+
+                headerRow_01.selectTitleInput().hasLimitMessage("128/256 caractères restant");
```

Comment:
it will be easier to follow if put line 101 after line 98
it means you input title for 1st row then check the limit character for 1st row
after that you input description for 2nd row then check the limit character for 2nd row

Reply:
and we can also make chain for these actions